sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -958,14 +1062,16 @@ object TypeCoercion extends TypeCoercionBase {
 
   override def implicitCast(e: Expression, expectedType: AbstractDataType): Option[Expression] = {
     implicitCast(e.dataType, expectedType).map { dt =>
-      if (dt == e.dataType) e else Cast(e, dt)
+      if (dt == e.dataType) { e }
```

Comment:
if the whole if statement isn't in one line statements shouldn't be on the same line as the condition
see https://github.com/databricks/scala-style-guide?tab=readme-ov-file#curly-braces

Reply:
I changed it for debugging, as I wanted to see which branch I jump into, will change it back.